base/promotion.jl

```diff
@@ -364,15 +364,10 @@ max(x::Real, y::Real) = max(promote(x,y)...)
 min(x::Real, y::Real) = min(promote(x,y)...)
 minmax(x::Real, y::Real) = minmax(promote(x, y)...)
 
-# "Promotion" that takes a function into account and tries to preserve
-# non-concrete types. These are meant to be used mainly by elementwise
-# operations, so it is advised against overriding them
-_default_type(T::Type) = T
-
 if isdefined(Core, :Compiler)
-    const _return_type = Core.Compiler.return_type
+    const _unsafe_return_type = Core.Compiler.return_type
```

Comment:
Technically we're allowed to rename this, but I believe several packages use it so we should continue to support the old name.